crates/agglayer-telemetry/src/traces.rs

```diff
@@ -0,0 +1,201 @@
+                    if let Ok(value) = max_attributes_per_link.parse::<u32>() {
+                        span_limits.max_attributes_per_link = value;
+                    } else {
+                        eprintln!("Failed to parse OTLP_MAX_ATTRIBUTES_PER_LINK");
+                    }
+                }
+                span_limits
+            };
+
+            let trace_provider = opentelemetry_sdk::trace::SdkTracerProvider::builder()
+                .with_sampler(Sampler::AlwaysOn)
+                .with_span_limits(span_limits)
+                .with_resource(Resource::builder().with_attributes(resources).build())
+                .with_span_processor(
+                    BatchSpanProcessor::builder(otlp_exporter)
+                        .with_batch_config(batch_processor_config.build())
+                        .build(),
+                )
+                .build();
+
+            let tracer = trace_provider.tracer("agglayer-otlp");
+
+            let _ = global::set_tracer_provider(trace_provider);
+
+            layers.push(
+                tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(
+                        EnvFilter::try_from_default_env().unwrap_or_else(|_| config.level.into()),
+                    )
+                    .boxed(),
+            );
+
+            global::set_text_map_propagator(TraceContextPropagator::new());
+        } else {
+            layers.push(match config.format {
+                TracingFormat::Pretty => tracing_subscriber::fmt::layer()
+                    .pretty()
+                    .with_writer(writer.as_make_writer())
+                    .with_filter(
+                        EnvFilter::try_from_default_env().unwrap_or_else(|_| config.level.into()),
+                    )
+                    .boxed(),
+
+                TracingFormat::Json => tracing_subscriber::fmt::layer()
+                    .json()
+                    .with_writer(writer.as_make_writer())
+                    .with_filter(
+                        EnvFilter::try_from_default_env().unwrap_or_else(|_| config.level.into()),
+                    )
+                    .boxed(),
+            });
+        }
+    }
+
+    // We are using try_init because integration test may try
+    // to initialize this multiple times.
+    tracing_subscriber::Registry::default()
+        .with(layers)
+        .try_init()
+        .map_err(|e| anyhow!("Unable to initialize tracing subscriber: {e:?}"))?;
+
+    tracing::info!("Tracing initialized with config: {config:?}");
+
+    Ok(())
+}
+
+fn build_resources(otlp_service_name: &str, version: &str) -> Vec<KeyValue> {
+    let mut resources = Vec::new();
+
+    resources.push(KeyValue::new("service.name", otlp_service_name.to_string()));
+    resources.push(KeyValue::new("service.version", version.to_string()));
+
+    let custom_resources: Vec<_> = std::env::var("AGGLAYER_OTLP_TAGS")
```

Comment:
Should it be scoped as `AGGLAYER_NODE_OTLP_TAGS` ?
Do we want those tags to be shared among other software?